holidays/utils.py

```diff
@@ -778,3 +778,394 @@ def thaipusam_date(self, year: int) -> date:
+                (On Month 8/8 for Athikamat Years).
+        - Buddhist Lent Day/Wan Khao Phansa:
+                1st Waning Day of Month 8
+                (On Month 8/8 for Athikamat Years).
+
+    Other Buddhist date on Thai Lunar Calendar:
+        - Buddha's Cremation Day/Atthami Bucha
+                8th Waning Day of  Month 6
+                (On Month 7 for Athikamat Years).
+        - End of Buddhist Lent Day/Ok Phansa:
+                15th Waxing Day (Full Moon) of Month 11
+
+    The following code is based on Ninenik Narkdee's PHP implementation,
+    and we're thankful for his work.
+
+    Please avoid touching the Athikawan and Athikamat declaration array
+    at all costs unless you can find sources for them somewhere for 2057++
+
+    Sources: (Ninenik.com 's wbm) http://tiny.cc/wa_ninenik_thluncal_php
+             https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2560.aspx
+
+    Usage example:
+
+    >>> from holidays.utils import _ThaiLuniSolar
+    >>> thls = _ThaiLuniSolar()
+    >>> print(thls.visakha_bucha_date(2010))
+    2010-05-28
+    """
+
+    # Athikawan (Extra-Day Year) list goes from 1941-2057 C.E.
+    # Copied off from 1757-2057 (B.E. 2300-2600) Thai Lunar Calendar
+    ATHIKAWAN_YEARS_GREGORIAN = {
+        1945,
+        1949,
+        1952,
+        1957,
+        1963,
+        1970,
+        1973,
+        1979,
+        1987,
+        1990,
+        1997,
+        2000,
+        2006,
+        2009,
+        2016,
+        2020,
+        2025,
+        2032,
+        2035,
+        2043,
+        2046,
+        2052,
+    }
+
+    # Athikamat (Extra-Month Year) list goes from 1941-2057 C.E.:
+    # Copied off from 1757-2057 (B.E. 2300-2600) Thai Lunar Calendar
+    # Approx formula as follows: (common_era-78)-0.45222)%2.7118886 < 1
+    ATHIKAMAT_YEARS_GREGORIAN = {
+        1942,
+        1944,
+        1947,
+        1950,
+        1953,
+        1956,
+        1958,
+        1961,
+        1964,
+        1966,
+        1969,
+        1972,
+        1975,
+        1977,
+        1980,
+        1983,
+        1985,
+        1988,
+        1991,
+        1994,
+        1996,
+        1999,
+        2002,
+        2004,
+        2007,
+        2010,
+        2012,
+        2015,
+        2018,
+        2021,
+        2023,
+        2026,
+        2029,
+        2031,
+        2034,
+        2037,
+        2040,
+        2042,
+        2045,
+        2048,
+        2050,
+        2053,
+        2056,
+    }
+
+    # While Buddhist Holy Days have been observed since the 1900s
+    #   Due to the calendar changes in 1941 (B.E. 2484) and that
+    #   our array only goes up to B.E. 2600; We'll thus only populate
+    #   the data for 1941-2057 (B.E. 2484-2600).
+    # Sources: หนังสือเวียนกรมการปกครอง กระทรวงมหาดไทย ที่ มท 0310.1/ว4
+    #          ออกเมื่อ 5 กุมภาพันธ์ พ.ศ.2539
+    START_DATE = date(1940, 11, 30)
+    START_YEAR = 1941
+    END_YEAR = 2057
+
+    @lru_cache
```

Comment:
It seems this syntax [is not acceptable](https://github.com/dr-prodigy/python-holidays/actions/runs/4143973021/jobs/7171773552#step:5:1) at least in py3.7
```suggestion
    @lru_cache()
```

Reply:
`@lru_cache()` does indeed seem to be the correct declaration rather than just `@lru_cache`